amida-tech/greyscale
# Notifications revamp

#### What's this PR do?
* Expanded notifications center
* Send notifications at each step of review
* Send a copy of notifications to current user

#### Related JIRA tickets:
https://jira.amida-tech.com/browse/GREY-615
https://jira.amida-tech.com/browse/GREY-619
https://jira.amida-tech.com/browse/GREY-623

#### How should this be manually tested?
@HarryQuid if you could take a quick glance through the code, that'd be great.

@ekavaliM take a look at the screenshots below. If they look good, I'll merge and you can QA on the CI deployment.

![image](https://cloud.githubusercontent.com/assets/574332/21203926/29361ac2-c255-11e6-8f0a-e6f27a417873.png)
The new Notifications Center. Top section shows notifications received _by_ the current user, both read and unread. Bottom section shows notifications _sent_ from the current user. If you're an organizational admin (unsure of the correct terminology -- whatever slots between admin and superadmin) you see all notifications for the organization, if you're a superadmin you see all notifications period.

![image](https://cloud.githubusercontent.com/assets/574332/21204027/80e53f64-c255-11e6-8074-d1f6835a2bf9.png)
When sending any kind of notification there's a "Send me a copy checkbox". If you check it, you'll actually receive the notification. If you don't check it, you'll still be able to see the sent notification in the bottom section of the notifications center.

![image](https://cloud.githubusercontent.com/assets/574332/21204043/a2c37a60-c255-11e6-9e7c-933187b6e309.png)
When moving a project onto the next step (by the force step arrow on the project dashboard, by the edit project modal on the projects listing, or the play/pause icons on the projects listing), you get a notification modal allowing you to send an optional message to all users involved. Same "Send me a copy" deal as with normal notifications.

#### Any background context you want to provide?
#### Screenshots (if appropriate):
